scaleway/resource_compute_instance_server.go

```diff
@@ -0,0 +1,297 @@
+	return &schema.Resource{
+		Create: resourceScalewayComputeInstanceServerCreate,
+		Read:   resourceScalewayComputeInstanceServerRead,
+		Update: resourceScalewayComputeInstanceServerUpdate,
+		Delete: resourceScalewayComputeInstanceServerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The name of the server",
+			},
+			"image": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "The base image of the server", // TODO: add in doc example with UUID or distrib name
+				ValidateFunc: validationUUID(),
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The instance type of the server", // TODO: link to scaleway pricing in the doc
+			},
+			"tags": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "The tags associated with the server",
+			},
+			"security_group_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The security group the server is attached to", // TODO: add this field in CreateServerRequest (proto)
+			},
+			//"root_volume": {
+			//	Type:     schema.TypeMap,
+			//	Optional: true,
+			//	ForceNew: true,
+			//	Elem: &schema.Resource{
+			//		Schema: map[string]*schema.Schema{
+			//			"size": {
+			//				Type:     schema.TypeString,
+			//				Optional: true,
+			//			},
+			//			"id": {
+			//				Type:     schema.TypeString,
+			//				Computed: true,
+			//			},
+			//		},
+			//	},
+			//	Description: "Root volume attached to the server on creation",
+			//},
+			"enable_ipv6": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "determines if IPv6 is enabled for the server",
+			},
+			"private_ip": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "the Scaleway internal IP address of the server", // TODO: add this field in CreateServerRequest (proto)
+			},
+			"public_ip": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "the public IPv4 address of the server",
+			},
+			"state": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "started",
+				Description: "the state of the server should be (started, stopped, standby)",
+				ValidateFunc: validation.StringInSlice([]string{
+					"started",
+					"stopped",
+					"standby",
+				}, false),
+			},
+			"user_data": {
+				Type: schema.TypeMap,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "The user data associated with the server", // TODO: document reserved keys (`cloud-init`)
```

Comment:
Maybe we should move cloud-init to a specific server property as this would trigger a forceNew

Reply:
True, it should ForceReboot instead of ForceNew though